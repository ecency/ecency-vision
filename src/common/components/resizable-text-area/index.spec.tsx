import React from "react";

import ResizableTextArea from "./index";
import renderer from "react-test-renderer";

const longText =
  "ResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextArea";
const props = {
  id: "the-editor-xs",
  className: "the-editor accepts-emoji form-control",
  minRows: 1,
  maxRows: 100,
  value: longText
};

const component = renderer.create(
  <div style={{ width: 300 }}>
    <ResizableTextArea {...props} />
  </div>
);

it("(1) Default render", () => {
  expect(component.toJSON()).toMatchSnapshot();
});

it("(2) With correct value", () => {
  const instance: any = component.getInstance();
  expect(component.root.findByProps({ id: props.id }).props.value).toContain(longText);
});
