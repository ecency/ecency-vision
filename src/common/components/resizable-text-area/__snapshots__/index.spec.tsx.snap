// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  style={
    Object {
      "width": 300,
    }
  }
>
  <textarea
    className="the-editor accepts-emoji form-control"
    id="the-editor-xs"
    maxRows={100}
    minRows={1}
    onChange={[Function]}
    rows={1}
    value="ResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextAreaResizableTextArea"
  />
</div>
`;
